docs/mkdocs/docs/integration/migration_guide.md

```diff
@@ -0,0 +1,248 @@
+          }
+          ```
+
+    === "Future-proof"
+
+          ```cpp
+          for (auto &x : j.items())
+          {
+              std::cout << x.key() << ":" << x.value() << std::endl;
+          }
+          ```
+
+- Function `friend std::ostream& operator>>(const basic_json&, std::ostream&)` is deprecated since 3.0.0. Please use
+  [`friend operator<<(std::ostream&, const basic_json&)`](../api/operator_ltlt.md) instead.
+  
+    === "Deprecated"
+
+          ```cpp
+          j >> std::cout;
+          ```
+
+    === "Future-proof"
+
+          ```cpp
+          std::cout << j;
+          ```
+
+- The legacy comparison behavior for discarded values is deprecated since 3.11.0. It is already disabled by default and
+  can still be enabled by defining
+  [`JSON_USE_LEGACY_DISCARDED_VALUE_COMPARISON`](../api/macros/json_use_legacy_discarded_value_comparison.md).
+  
+    === "Deprecated"
+
+          ```cpp
+          #define JSON_USE_LEGACY_DISCARDED_VALUE_COMPARISON
+          #include <nlohmann/json.hpp>
+          ```
+
+    === "Future-proof"
+
+          ```cpp
+          #include <nlohmann/json.hpp>
+          ```
+
+## Replace implicit conversions
+
+Implicit conversions via [`operator ValueType`](../api/basic_json/operator_ValueType.md) will be switched off by default
+in the next major release of the library.
+
+=== "Deprecated"
+
+      ```cpp
+      nlohmann::json j = "Hello, world!";
+      std::string s = j;
+      ```
+
+=== "Future-proof"
+
+      ```cpp
+      nlohmann::json j = "Hello, world!";
+      auto s = j.get<std::string>();
+      ```
+
+You can prepare existing code by already defining
+[`JSON_USE_IMPLICIT_CONVERSIONS`](../api/macros/json_use_implicit_conversions.md) to `0` and replace any implicit
+conversions with calls to [`get`](../api/basic_json/get.md).
+
+## Import namespace `literals` for UDLs
+
+The user-defined string literals [`operator""_json`](../api/operator_literal_json.md) and
+[`operator""_json_pointer`](../api/operator_literal_json_pointer.md) will be removed from the global namespace in the
+next major release of the library.
+
+=== "Deprecated"
+
+      ```cpp
+      nlohmann::json j = "[1,2,3]"_json;
+      ```
+
+=== "Future-proof"
+
+      ```cpp
+      using namespace nlohmann::literals;
+      nlohmann::json j = "[1,2,3]"_json;
+      ```
+
+To prepare existing code, define [`JSON_USE_GLOBAL_UDLS`](../api/macros/json_use_global_udls.md) to `0` and bring the
+string literals into scope where needed.
+
+## Do not hard-code the complete library namespace
+
+The [`nlohmann` namespace](../features/namespace.md) contains two sub-namespaces to avoid problems when different
+versions or configurations of the library are used in the same project. Always use `nlohmann` as namespace or, when the
+exact version and configuration is relevant, use macro
+[`NLOHMANN_JSON_NAMESPACE`](../api/macros/nlohmann_json_namespace.md) to denote the namespace.
```

Comment:
This section is tricky. I keep oscillating on what is necessary and when.

I'll expand on this later.

(Also, "two sub-namespaces" is definitely wrong now.)

Reply:
I fixed the "two sub-namespaces". No idea about the rest.

Reply:
I hesitate because of the advice given here:
https://json.nlohmann.me/features/arbitrary_types/#how-do-i-convert-third-party-types

Currently, I think this section is OK (i.e., the issue is resolved for the purpose of this PR) but I'm conflicted about the advice to use `NLOHMAN_JSON_NAMESPACE_BEGIN` when specializing `adl_serializer`.